### PR TITLE
doc: rgw: Rewrite the key management

### PR DESCRIPTION
**Pull request by hrchu**
The create/remove key section in the doc is a bit confusing. (Especially for swift/s3 subuser)
Rewrite it and give more examples.

### CODE REVIEW

File: doc/radosgw/admin.rst [hunk 1]
```diff
@@ -211,61 +211,104 @@ Options include:
...
-        "secret_key": "iaSFLDVvDdQt6lkNzHyW4fPLZugBAI1g17LO0+87"}],
     "swift_keys": [
-      { "user": "johndoe:swift",
-        "secret_key": "E9T2rUZNu2gxUjcwUBO8n\/Ev4KX6\/GprEuH4qhu1"}]}
+      { "user": "foo:bar",
+        "secret_key": "asfghjghghmgm"}]}
 
+You may also specify the secret.::
 
+	radosgw-admin key create --subuser=foo:bar --gen-access-key
```
Comment (cbodley) [inline doc/radosgw/admin.rst]: i'm not clear on this part, the command doesn't specify a secret

also, the `secret.::` part is rendered as `secret.:` - it should be `secret. ::`

File: doc/radosgw/admin.rst [hunk 2]
```diff
@@ -211,61 +211,104 @@ Options include:
...
-------------------------
+Note that a subuser can only have one swift secret.
 
-Users and subusers must have access keys to use the S3 and Swift
-interfaces. When you create a user or subuser and you do not specify 
-an access key and secret, the key and secret get generated automatically. 
-You may create a key and either specify or generate the access key and/or
-secret. You may also remove an access key and secret. Options include:
+In the least radosgw version, subuser can also operates S3 APIs. To create a S3
+key pair for a subuser. ::
```
Comment (cbodley) [inline doc/radosgw/admin.rst]: > In the least radosgw version

documentation doesn't need to refer to versions. this documentation going into the `master` branch, so you can assume that all features are present

> subuser can also operates S3 APIs

maybe clarify this with `Subusers can also be used with S3 APIs.`

File: doc/radosgw/admin.rst [hunk 3]
```diff
@@ -211,61 +211,104 @@ Options include:
...
 
+	radosgw-admin key create --subuser=foo:bar --key-type=s3 --gen-access-key
 
-- ``--secret=<key>`` specifies a secret key (e.g,. manually generated).
-- ``--gen-access-key`` generates random access key (for S3 user by default).
-- ``--gen-secret`` generates a random secret key.
-- ``--key-type=<type>`` specifies a key type. The options are: swift, s3
+or sepficy the key pair. ::
```
Comment (cbodley) [inline doc/radosgw/admin.rst]: the `--gen-access-key` flag was already described above, so i think it's redundant to give both examples (`--gen-access-key` and `--access-key`/`--secret-key`) for subusers also. let's just keep this one example with `--access-key barAccessKey --secret-key barSecretKey` so it matches the json output below

File: doc/radosgw/admin.rst (hunk 1 above)
Comment (hrchu) [inline doc/radosgw/admin.rst]: Fixed

File: doc/radosgw/admin.rst (hunk 2 above)
Comment (hrchu) [inline doc/radosgw/admin.rst]: Done

File: doc/radosgw/admin.rst (hunk 3 above)
Comment (hrchu) [inline doc/radosgw/admin.rst]: Roger roger

**Comment by hrchu** [comment]
Revised according to @cbodley 's comments.

**Comment by cbodley** [comment]
jenkins test this please

**Pull request merged**